qiskit/primitives/containers/data_bin.py

```diff
@@ -15,78 +15,78 @@
 """
 from __future__ import annotations
 
-from collections.abc import Iterable
-from dataclasses import make_dataclass
+from typing import Any, Iterable
 
 
 class DataBinMeta(type):
-    """Metaclass for :class:`DataBin` that adds the shape to the type name.
+    """This metaclass no longer has a purpose and will be removed."""
 
-    This is so that the class has a custom repr with DataBin<*shape> notation.
-    """
 
-    def __repr__(cls):
-        name = cls.__name__
-        if cls._SHAPE is None:
-            return name
-        shape = ",".join(map(str, cls._SHAPE))
-        return f"{name}<{shape}>"
+class DataBin(metaclass=DataBinMeta):
+    """Namespace for storing data.
```

Comment:
Should this inherit from `Mapping` now? Also its now both a namespace and a mapping container

Reply:
The iter already exists from a different PR. I'd like to control our own namespace, so I think I'm happy to be a mapping only via protocol.